portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,40 +33,49 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  
-  private List<String> comments;
-
-  @Override
-  public void init() {
-    comments = new ArrayList<>();
-  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    int numComm = Integer.parseInt(request.getParameter("num-choice"));
+    
+    PreparedQuery results = datastore.prepare(query);
+
+    //loading comments from Datastore
+    List<String> comments = new ArrayList<>();
+    for (Entity commentEntity : results.asIterable(FetchOptions.Builder.withLimit(numComm))) {
+      String txt = (String) commentEntity.getProperty("text-input");
+      comments.add(txt);
+    }
+
```

Comment:
This code is all from week 3.  You should update your branch to pull these changes in.